protostar/commands/init/project_creator/new_project_creator.py

```diff
@@ -52,12 +66,85 @@ def _gather_input(self) -> "NewProjectCreator.NewProjectConfig":
         cairo_version: CairoVersion,
     ) -> None:
         output_dir_path = self._output_dir_path
-        project_root_path = output_dir_path / user_input.project_dirname
-        self.copy_template(cairo_version, project_root_path)
-        self.save_protostar_toml(
-            project_root_path=project_root_path, cairo_version=cairo_version
+        project_root_path = output_dir_path / project_config.project_dirname
+        project_name = project_config.project_dirname
+
+        self.copy_and_replace_template(
+            cairo_version=cairo_version,
+            project_root_path=project_root_path,
+            project_name=project_name,
+        )
+        self._save_protostar_toml(
+            project_root_path=project_root_path,
+            configuration_model=self._make_project_config(cairo_version=cairo_version),
+        )
+
+    def copy_and_replace_template(
+        self, cairo_version: CairoVersion, project_root_path: Path, project_name: str
+    ):
+        template_path = self.script_root / "templates" / cairo_version.value
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_dir = Path(temp_dir)
+            new_template_path = Path(temp_dir) / "template"
+
+            shutil.copytree(template_path, new_template_path)
+            self._insert_project_name_to_template(project_name, new_template_path)
+
+            try:
+                shutil.copytree(new_template_path, project_root_path)
+            except FileExistsError as ex_file_exists:
+                raise ProtostarException(
+                    f"Folder or file named {project_root_path.name} already exists. Choose different project name."
+                ) from ex_file_exists
+
+    def _make_project_config(
+        self, cairo_version: CairoVersion
+    ) -> ConfigurationFileV2Model:
+        if cairo_version == CairoVersion.cairo0:
+            return ConfigurationFileV2Model(
+                protostar_version=str(self._protostar_version),
+                contract_name_to_path_strs={"main": ["src/main.cairo"]},
+                project_config={
+                    "lib-path": "lib",
+                },
+                command_name_to_config={},
+                profile_name_to_project_config={},
+                profile_name_to_commands_config={},
+            )
+
+        return ConfigurationFileV2Model(
+            protostar_version=str(self._protostar_version),
+            contract_name_to_path_strs={"hello_starknet": ["src"]},
+            project_config={
+                "lib-path": "lib",
+                "linked-libraries": ["src"],
+            },
+            command_name_to_config={},
+            profile_name_to_project_config={},
+            profile_name_to_commands_config={},
         )
+
+    @staticmethod
+    def _insert_project_name_to_template(project_name: str, temp_dir: Path):
+        for path in temp_dir.glob("**/*"):
+            if not path.is_file():
+                continue
```

Comment:
This is a bit aggressive, can we predefine the paths?

Reply:
It's only iterating through a temp directory with template, I think it should be fine? 

We could predefine paths but then they would have to be updated on every template change.

Reply:
don't have a strong opinion, it'd be easier to understand though. do as you see fit in this case

Reply:
cc @MaksymilianDemitraszek @karol-bisztyga @mkaput

Reply:
I think it's ok to do this, pretty straightforward. I'm not sure what is the possible problem here.

Reply:
> This doesn't look to be super fragile and could stay IMHO, but I am still more in favour of just storing files contents as strings and avoid this temporary directory fluff

Using temp directories here is more straightforward I think because we don't have to manually recreate the directory structure from template - we just copy it.